packages/signers/src/transaction-modifying-signer.ts

```diff
@@ -0,0 +1,29 @@
+import { Address } from '@solana/addresses';
+import { CompilableTransaction } from '@solana/transactions';
+
+/** Defines a signer capable of signing transactions. */
+export type TransactionModifyingSigner<TAddress extends string = string> = Readonly<{
+    address: Address<TAddress>;
+    modifyAndSignTransaction<TTransaction extends CompilableTransaction>(
```

Comment:
Same here on the dictionary return value.

Reply:
Here as well the returned transaction already contains a `SignatureDictionary` in its `signatures` attribute.